plugins/out_azure_blob/azure_blob.c

```diff
@@ -420,7 +428,6 @@ static int send_blob(struct flb_config *config,
         /* For Logs type, we need to commit the block right away */
         if (event_type == FLB_EVENT_TYPE_LOGS) {
             ret = azb_block_blob_commit_block(ctx, block_id, tag, ms);
-            flb_free(block_id);
```

Comment:
Isn't it possible, that only removing this line should be enough? Calling `flb_free` on NULL should not be a problem.

Reply:
Just removing that line fixes the double free but considering that there are code paths that lead to calling `free` with `block_id` holding `NULL` it's worth adding the check since older systems (and maybe alternative libc implementations?) don't handle it gracefully.